scroll highlighted file names on SD card #5

armin-fx opened the pull request.

armin-fx committed:
scroll highlighted file names on SD card

armin-fx committed:
scroll highlighted file names on SD card

armin-fx committed:
change size of buffer to max filename length

armin-fx committed:
show more from long filenames (26 to 54 chars)

use 182 byte more ram

armin-fx committed:
scroll highlighted file names on SD card

armin-fx committed:
scroll highlighted file names on SD card

can write anywhere when buffer is smaller then displayable text

TinkerGnome commented:
Hi armin-fx,
this change increases the size of the lcd_cache by a significant amount, and we are already short on RAM. Do you see a chance to avoid this?

armin-fx committed:
Merge pull request #1 from TinkerGnome/geek_mode

update this fork

armin-fx committed:
change lcd_cache size

the file name cache is now split in two parts. first part will store only the first 20 direct displayable letter.
if one file name will be selected (file name will be scrolled), full file name will be loaded from SD card and the remain of the file name will store into the smaller second part.
(only one file must be loaded with fully name.)

armin-fx committed:
change lcd_cache size

the file name cache is now split in two parts. first part will store only the first 20 direct displayable letter.
if one file name will be selected (file name will be scrolled), full file name will be loaded from SD card and the remain of the file name will store into the smaller second part.
(only one file must be loaded with fully name.)

armin-fx commented:
Hi TinkerGnome,
sorry for this late answer.

Now I have split lcd_cache file name buffer split in two parts.
The first part store only the first 20 direct displayable letter which must not scrolled.
The second smaller part will be loaded, when needed and store the rest of the file name.

What do you mean about (may slow) scrolling long file name (or long print message) in print menu when printer is printing?

armin-fx committed:
Merge pull request #2 from TinkerGnome/geek_mode

merge TinkerGnome/geek_mode into this branch

armin-fx committed:
scroll displayed file name when printing

armin-fx committed:
Update UltiLCD2_hi_lib.h

armin-fx committed:
Merge pull request #3 from TinkerGnome/geek_mode

merge TinkerGnome/geek_mode into this branch

TinkerGnome commented:
finally I found some time to [merge it](https://github.com/TinkerGnome/Ultimaker2Marlin/commit/e1a20c36e3e477b7caf1c420e02cb83049759933)  😃
